src/packages/version-check/src/index.ts

```diff
@@ -0,0 +1,291 @@
+  }
+
+  setUrl(url) {
+    this.set("url", url);
+  }
+
+  private set(key, value) {
+    this._config[key] = value;
+    this.saveConfig();
+  }
+
+  private saveConfig() {
+    this.ConfigManager.set("config", this._config);
+  }
+
+  configFileLocation() {
+    return this.ConfigManager.path;
+  }
+
+  // Intentionally verbose here if we get logging involved it could aid debugging
+  canNotifyUser() {
+    const currentVersion = this._currentVersion;
+    const latestVersion = this._config.latestVersion;
+    // No currentVersion version passed in
+    if (!currentVersion) {
+      return false;
+    } else if (this.alreadyLoggedThisVersion()) {
+      return false;
+    }
+    // returns falsy if function cannot detect semver difference
+    return !!this.detectSemverChange(currentVersion, latestVersion);
+  }
+
+  detectSemverChange(currentVersion, latestVersion) {
+    if (
+      !currentVersion ||
+      !latestVersion ||
+      !this.isValidSemver(currentVersion) ||
+      !this.isValidSemver(latestVersion) ||
+      semverGte(currentVersion, latestVersion)
+    )
+      return null;
+
+    return semverDiff(currentVersion, latestVersion);
+  }
+
+  async getLatestVersion() {
+    if (!this._config.enabled) return false;
+    try {
+      const latestVersion = this.cleanSemver(await this.fetchLatestVersion());
+      this.setLatestVersion(latestVersion);
+      return true;
+    } catch (e) {
+      // The fail is silent
+      return false;
+    }
+  }
+
+  private fetchLatestVersion() {
+    const { packageName, url, ttl } = this._config;
+
+    return new Promise<string>((resolve, reject) => {
+      const session = http2.connect(url);
+
+      session.on("error", err => reject(err));
+
+      const req = session.request({ ":path": `/?name=${packageName}` });
+
+      req.setEncoding("utf8");
+
+      req
+        .on("error", reject)
+        .on("response", (headers, flags) => {
+          let data = "";
+          req
+            .on("data", chunk => {
+              data += chunk;
+            })
+            .on("end", async () => {
+              resolve(data);
+              session.close();
+            });
+        })
+        .setTimeout(ttl, () => {
+          req.close();
+          session.close();
+          reject();
+        });
+      req.end();
+    });
+  }
+
+  alreadyLoggedThisVersion() {
+    return semverGte(
+      this._config.latestVersionLogged,
+      this._config.latestVersion
+    );
+  }
+
+  log() {
+    if (!this._config.enabled || !this.canNotifyUser()) return false;
+
+    const currentVersion = this._currentVersion;
+    const { packageName, latestVersion } = this._config;
+
+    const upgradeType = this.detectSemverChange(currentVersion, latestVersion);
+
+    this.logBannerMessage({
+      upgradeType,
+      packageName,
+      currentVersion,
+      latestVersion
+    });
+    this.setLatestVersionLogged(latestVersion);
+    return true;
+  }
+
+  logBannerMessage(options) {
```

Comment:
Nice one.

:+1: I was able to reduce some unit tests and some redundant checks since it is private now https://github.com/trufflesuite/ganache/pull/3285/commits/65559e784a56267518056102af87b5635b1e6627